app/forms/questionnaire_form.py

```diff
@@ -69,9 +134,12 @@ def map_errors(self):
     def answer_errors(self, input_id):
         return [error[1] for error in self.map_errors() if input_id == error[0]]
 
-    def get_data(self, answer_id):
+    def get_data(self, answer_id, default=SENTINEL):
```

Comment:
I don't understand what SENTINEL is??

Reply:
its a flag value, it helps when there is `none` data for a value, i.e stops the loop

Reply:
Yeah it's just so you're able to pass `(default=None)` as an argument. It'd work to have a `default='some_other_value'` but sentinel is a posher way to define it 😃

Reply:
Why cant we just default it to none?
`def get_data(self, answer_id, default=None):`

Reply:
Because the current behaviour is to raise an `AttributeError` if `get_data` is called without a default. Changing that would change its behaviour and require all callers to handle `None` values.

It's just a pattern to maintain a sensible interface and not allow errors to cascade to other areas in the system.

Reply:
I don't like it, i think it makes the code hard to read.
I feel it compromises code readability.

Reply:
I disagree

Reply:
@shahi645 I don't understand this either so could you explain it to us please.

Reply:
@shahi645 I changed this to the following and it seems to work
```
    def get_data(self, answer_id, default=None):
        attr = getattr(self, answer_id)
        value = attr.raw_data[0] if attr.raw_data else ''
        if default is not None and not value:
            return default
        return value
```

Reply:
I'll remove the SENTINEL value 😖